src/world/partner/sushie.c

```diff
@@ -1,42 +1,150 @@
 #include "common.h"
+#include "../src/world/partners.h"
+#include "npc.h"
 
 extern unkPartnerStruct* D_802BFDF8_320B68;
+extern s32 D_802BFEEC;
+extern f32 D_802BFEE0;
+extern void func_802BD368_31E0D8(s32, f32, f32, f32, f32, f32);
+extern void playFX_23(s32, f32, f32, f32, s32);
+extern s32 partner_use_ability(void);
+extern void partner_kill_ability_script(void);
+extern s32 D_802BFEE4;
+extern s32 D_802BFEE8;
+extern s32 D_802BFEF0;
+extern s32 D_802BFEF4;
+extern f32 D_802BFDB0_320B20;
+extern f32 D_802BFDB4_320B24;
+extern s16 D_8010C97A;
 
-INCLUDE_ASM(s32, "world/partner/sushie", func_802BD100_31DE70);
 
-INCLUDE_ASM(s32, "world/partner/sushie", func_802BD20C_31DF7C);
+void func_802BD100_31DE70(void) {
+    Npc* partnerNPC = get_npc_unsafe(NPC_PARTNER);
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Camera* cam =  gCameras;
```

Comment:
&gCameras[0] (minor gripe but it's more explicit that we're getting the first one)